index.test-d.ts

```diff
@@ -6,30 +6,125 @@ import { barPlugin } from "./plugins/bar/index.js";
 import { voidPlugin } from "./plugins/void/index.js";
 import { withOptionsPlugin } from "./plugins/with-options";
 
-const base = new Base();
+const base = new Base({
+  version: "1.2.3",
+});
 
 // @ts-expect-error unknown properties cannot be used, see #31
 base.unknown;
 
-const FooBase = Base.plugin(fooPlugin).defaults({
-  default: "value",
+const BaseWithEmptyDefaults = Base.defaults({
+  // there should be no required options
+});
+
+// 'version' is missing and should still be required
+// @ts-expect-error
+new BaseWithEmptyDefaults()
+
+// 'version' is missing and should still be required
+// @ts-expect-error
+new BaseWithEmptyDefaults({})
+
+const BaseLevelOne = Base.plugin(fooPlugin).defaults({
+  defaultOne: "value",
+  version: "1.2.3",
+});
+
+// Because 'version' is already provided, this needs no argument
+new BaseLevelOne();
+new BaseLevelOne({});
+
+expectType<{
+  defaultOne: string,
+  version: string,
+}>(BaseLevelOne.defaultOptions);
+
+const baseLevelOne = new BaseLevelOne({
+  optionOne: "value",
+});
+
+expectType<string>(baseLevelOne.options.defaultOne);
+expectType<string>(baseLevelOne.options.optionOne);
+expectType<string>(baseLevelOne.options.version);
+// @ts-expect-error unknown properties cannot be used, see #31
+baseLevelOne.unknown;
+
+const BaseLevelTwo = BaseLevelOne.defaults({
+  defaultTwo: 0,
+});
+
+expectType<{
+  defaultOne: string,
+  defaultTwo: number,
+  version: string,
+}>({ ...BaseLevelTwo.defaultOptions });
```

Comment:
why the destructoring? Isn't this the same as 

```suggestion
}>(BaseLevelTwo.defaultOptions);
```

Reply:
Ha, this one was nifty. `tsd` sees the an intersection type as not being the same as its equivalent solo type.

```
  index.test-d.ts:86:0
  ✖  86:0  Parameter type { defaultOne: string; defaultTwo: number; defaultThree: string[]; version: string; } is not identical to argument type { defaultOne: string; version: string; } & { defaultTwo: number; } & { defaultThree: string[]; }.  
```